### PR TITLE
support init_frz_model for hybrid descriptor

### PR DESCRIPTION
**Pull request by njzjz**
Refactors some methods to implement it.
Also fixes some typos.

### CODE REVIEW

**Review by amcadmus — APPROVED** [review]

File: deepmd/descriptor/se_a.py [hunk 1]
```diff
@@ -481,60 +496,21 @@ def pass_tensors_from_frz_model(self,
...
-            't_natoms:0' :natoms,
-            't_box:0'    :box,
-            't_mesh:0'   :mesh
-        }
-        return feed_dict
-
-
     def init_variables(self,
-                       embedding_net_variables: dict
+                       graph_def : tf.GraphDef,
```
Comment (denghuilu) [inline deepmd/descriptor/se_a.py:500]:
```suggestion
                       model_file : str,
```

File: deepmd/descriptor/se_a.py [hunk 2]
```diff
@@ -481,60 +496,21 @@ def pass_tensors_from_frz_model(self,
...
     ) -> None:
         """
         Init the embedding net variables with the given dict
 
         Parameters
         ----------
-        embedding_net_variables
-                The input dict which stores the embedding net variables
+        graph_def : tf.GraphDef
+            The input tf.GraphDef object
```
Comment (denghuilu) [inline deepmd/descriptor/se_a.py:509]:
```suggestion
        model_file : str
            The input frozen model file
```

File: deepmd/descriptor/se_a.py [hunk 3]
```diff
@@ -481,60 +496,21 @@ def pass_tensors_from_frz_model(self,
...
+        suffix : str, optional
+            The suffix of the scope
         """
-        self.embedding_net_variables = embedding_net_variables
+        self.embedding_net_variables = get_embedding_net_variables(graph_def, suffix = suffix)
```
Comment (denghuilu) [inline deepmd/descriptor/se_a.py:513]:
```suggestion
        self.embedding_net_variables = get_embedding_net_variables(model_file, suffix = suffix)
```

File: deepmd/descriptor/se_a.py (hunk 1 above)
Comment (denghuilu) [reply deepmd/descriptor/se_a.py:500]: The input parameter `self.run_opt.init_frz_model` is a model file

**Review by denghuilu — CHANGES_REQUESTED** [review]

File: deepmd/descriptor/hybrid.py [hunk 4]
```diff
@@ -253,3 +253,55 @@ def enable_compression(self,
...
+                       graph_def : tf.GraphDef,
+                       suffix : str = "",
+    ) -> None:
+        """
+        Init the embedding net variables with the given dict
+
+        Parameters
+        ----------
+        graph_def : tf.GraphDef
+            The input tf.GraphDef object
```
Comment (njzjz) [inline deepmd/descriptor/hybrid.py:267]:
```suggestion
        model_file : str
            The input frozen model file
```

File: deepmd/descriptor/hybrid.py [hunk 5]
```diff
@@ -253,3 +253,55 @@ def enable_compression(self,
         """
         for idx, ii in enumerate(self.descrpt_list):
             ii.enable_compression(min_nbor_dist, model_file, table_extrapolate, table_stride_1, table_stride_2, check_frequency, suffix=f"{suffix}_{idx}")
+
+    def init_variables(self,
+                       graph_def : tf.GraphDef,
```
Comment (njzjz) [inline deepmd/descriptor/hybrid.py:258]:
```suggestion
                       model_file : str,
```

File: deepmd/descriptor/hybrid.py [hunk 6]
```diff
@@ -253,3 +253,55 @@ def enable_compression(self,
...
+
+        Parameters
+        ----------
+        graph_def : tf.GraphDef
+            The input tf.GraphDef object
+        suffix : str, optional
+            The suffix of the scope
+        """
+        for idx, ii in enumerate(self.descrpt_list):
+            ii.init_variables(graph_def, suffix=f"{suffix}_{idx}")
```
Comment (njzjz) [inline deepmd/descriptor/hybrid.py:272]:
```suggestion
            ii.init_variables(model_file, suffix=f"{suffix}_{idx}")
```

File: deepmd/descriptor/descriptor.py [hunk 7]
```diff
@@ -280,26 +280,65 @@ def get_feed_dict(self,
...
+            't_natoms:0' :natoms,
+            't_box:0'    :box,
+            't_mesh:0'   :mesh
+        }
+        return feed_dict
 
     def init_variables(self,
-                       embedding_net_variables: dict
-                       ) -> None:
+                       graph_def : tf.GraphDef,
```
Comment (njzjz) [inline deepmd/descriptor/descriptor.py:293]:
```suggestion
                       model_file: str,
```

File: deepmd/descriptor/descriptor.py [hunk 8]
```diff
@@ -280,26 +280,65 @@ def get_feed_dict(self,
...
+                       suffix : str = "",
+    ) -> None:
         """
         Init the embedding net variables with the given dict
 
         Parameters
         ----------
-        embedding_net_variables
-                The input dict which stores the embedding net variables
+        graph_def : tf.GraphDef
```
Comment (njzjz) [inline deepmd/descriptor/descriptor.py:301]:
```suggestion
        model_file : str
```

File: deepmd/descriptor/descriptor.py [hunk 9]
```diff
@@ -280,26 +280,65 @@ def get_feed_dict(self,
...
+            The input tf.GraphDef object
```
Comment (njzjz) [inline deepmd/descriptor/descriptor.py:302]:
```suggestion
            The input model file
```

**Review by denghuilu — APPROVED** [review]

**Pull request merged**